TFNetworkRecLayer.py

```diff
@@ -943,30 +942,25 @@ def _construct(self, prev_outputs, prev_extra, i,
     :param dict[str,tf.Tensor] prev_outputs: outputs of the layers from the previous step
     :param dict[str,dict[str,tf.Tensor]] prev_extra: extra output / hidden states of the previous step for layers
     :param tf.Tensor i: loop counter. scalar, int32, current step (time)
-    :param tf.Tensor|None data: optional source data, shape e.g. (batch,dim)
-    :param tf.Tensor|None classes: optional target classes, shape e.g. (batch,) if it is sparse
+    :param dict[str,tf.Tensor] data: All data needed from outside of the loop. Possible keys are 'source'
+        (for the input of the recurrent layer) and the keys in parent_net.extern_data, which notably may include
+        the target of the recurrent layer.
```

Comment:
This is the updated description, is it clear now?

Reply:
I would mention `'classes'` as an example also.